servicefabric.go

```diff
@@ -41,9 +41,14 @@ type Provider struct {
 	AppInsightsInterval   flaeg.Duration   `description:"The interval for sending data to Application Insights, optional"`
 }
 
+// Init the provider
+func (p *Provider) Init(constraints types.Constraints) error {
+	return p.BaseProvider.Init(constraints)
```

Comment:
Would it make sense to move the top section of the `Provide` method into the `Init` method so we can return an error if the connection to SF isn't valid? For example:

```
 	sfClient, err := sf.NewClient(http.DefaultClient, p.ClusterManagementURL, p.APIVersion, tlsConfig)
``